props/saber_fett263_buttons.h

```diff
@@ -5593,22 +5598,23 @@ SaberFett263Buttons() : PropBase() {}
         clash_type_ = CLASH_BATTLE_MODE;
         return true;
 
-
       case EVENTID(BUTTON_NONE, EVENT_STAB, MODE_ON):
         if (menu_ || SaberBase::Lockup() || CheckShowColorCC()) return true;
         clash_impact_millis_ = millis();
-        if (!battle_mode_) {
-#ifdef FETT263_CLASH_STRENGTH_SOUND
-          clash_impact_millis_ = millis();
-          clash_type_ = CLASH_STAB;
-#else
-          SaberBase::DoStab();
-#endif
-          return true;
-        }
         check_blast_ = false;
         swing_blast_ = false;
         if (!swinging_) {
+#ifndef FETT263_USE_BC_MELT_STAB
```

Comment:
If I move then I'd have to repeat the "if (!swinging_) {" as #else, it needs to be there whether you use the define or not, it's part of the overall control.  The part that is affected by the define in only the "if (!battle_mode_) {...}".  I thought you didn't want me to repeat code if it's not affected by the define.

Reply:
There is an "if (!swinging_) {" right before the endif, just move that one out, and then it should be fine, right?

repeating code is bad (because you have to read and compare the code to understand what has changed)
but having ifdefs intersect control structures is also bad (because it is hard to read and understand the code)

Reply:
ok, I've updated

Reply:
I think you changed it in the opposite way of what I was looking for.  (Unless I'm missing something)

In pseudocode here is what it originally said:
if (!swinging) {
#ifdef DEF
   CODEA
}  
if (!swinging) {
#endif
   CODEB
}

Here is what you changed it to:

#ifdef DEF
if (!swinging) {
  CODEA
#else
if (!swinging) {
#endif
  CODEB
}

Here is how I think it should be written:
#ifdef DEF
if (!swinging) {
  CODEA
}
#endif
if (!swinging) {
  CODEB
}

Reply:
Not following.  
#ifndef FETT263_USE_BC_MELT_STAB

just adds 

if (!battle_mode_) {...} 

into 
if (!swinging_){...}

That's why I had the "if (!swinging_) {" outside of the define originally.   The close for line 5607 is at 5627, maybe you were seeing the close for the battle mode check?

Reply:
I'm super confused now.
When I look back at the various changes in here, I can't find the way I thought looked originally.
Now when I look, the second version looks fine. (Not the first first version, where the #ifdefs were too big)
We should go back to putting the ifdef inside the if statement I guess.
Not sure what happened, but I apologize for the confusion.

Reply:
no worries, I changed it back